package com.lineage.server.model.skill.skillmode;

import com.lineage.server.model.L1Character;
import com.lineage.server.model.L1Magic;
import com.lineage.server.model.Instance.L1NpcInstance;
import com.lineage.server.model.Instance.L1PcInstance;
import com.lineage.server.serverpackets.S_OwnCharAttrDef;
import com.lineage.server.serverpackets.S_SPMR;

public class DS_ASX08 extends SkillMode {
	private static final int _addac = -5;
	private static final int _addmr = 15;
	private static final int _adddmgdown = 2;
	private static final int _addstun = 2;

	public int start(L1PcInstance srcpc, L1Character cha, L1Magic magic, int integer) throws Exception {
		int dmg = 0;
		if (!srcpc.hasSkillEffect(4538)) {
			srcpc.addAc(-5);
			srcpc.addMr(15);
			srcpc.addDamageReductionByArmor(2);
			// srcpc.addRegistStun(2); // 昏迷耐性
			srcpc.setSkillEffect(4538, integer * 1000);
			srcpc.sendPackets(new S_OwnCharAttrDef(srcpc));
			srcpc.sendPackets(new S_SPMR(srcpc));
		}
		return dmg;
	}

	public int start(L1NpcInstance npc, L1Character cha, L1Magic magic, int integer) throws Exception {
		int dmg = 0;

		return dmg;
	}

	public void start(L1PcInstance srcpc, Object obj) throws Exception {
	}

	public void stop(L1Character cha) throws Exception {
		cha.addAc(5);
		cha.addMr(-15);
		// cha.addRegistStun(-2); // 昏迷耐性
		if ((cha instanceof L1PcInstance)) {
			L1PcInstance pc = (L1PcInstance) cha;
			pc.addDamageReductionByArmor(-2);
			pc.sendPackets(new S_OwnCharAttrDef(pc));
			pc.sendPackets(new S_SPMR(pc));
		}
	}
}
